src/main/java/de/tum/in/www1/artemis/service/connectors/localci/buildagent/LocalCIDockerService.java

```diff
@@ -246,15 +255,103 @@ public void pullDockerImage(LocalCIBuildJobQueueItem buildJob, BuildLogsMap buil
 
     @Scheduled(cron = "${artemis.continuous-integration.image-cleanup.cleanup-schedule-time:0 0 3 * * *}")
     public void deleteOldDockerImages() {
-
         if (!imageCleanupEnabled) {
             log.info("Docker image cleanup is disabled");
             return;
         }
 
+        Set<String> imageNames = getUnusedDockerImages();
+
         // Get map of docker images and their last build dates
         IMap<String, ZonedDateTime> dockerImageCleanupInfo = hazelcastInstance.getMap("dockerImageCleanupInfo");
 
+        // Delete images that have not been used for more than imageExpiryDays days
+        for (String dockerImage : dockerImageCleanupInfo.keySet()) {
+            if (imageNames.contains(dockerImage)) {
+                if (dockerImageCleanupInfo.get(dockerImage).isBefore(ZonedDateTime.now().minusDays(imageExpiryDays))) {
+                    log.info("Deleting docker image {}", dockerImage);
+                    try {
+                        dockerClient.removeImageCmd(dockerImage).exec();
+                    }
+                    catch (NotFoundException e) {
+                        log.warn("Docker image {} not found during cleanup", dockerImage);
+                    }
+                }
+            }
+        }
+    }
+
+    /**
+     * Checks for available disk space and triggers the cleanup of old Docker images if the available space falls below a specified threshold.
+     * The threshold is configurable via the 'artemis.continuous-integration.image-cleanup.disk-space-threshold-mb' property in the application settings.
```

Comment:
I think that when `imageCleanupDiskSpaceThresholdMb` is linked via Javadoc there is no need to explain that the variable can be configured via `artemis.continuous-integration.image-cleanup.disk-space-threshold-mb`.
In my opinion it is better to user the code as documentation there, as I think that the comment is not very likely to be updated if the setting name is updated

```suggestion
     * Checks for available disk space and triggers the cleanup of old Docker images if the available space falls below  {@link imageCleanupDiskSpaceThresholdMb}.
```

Reply:
I agree, changed it 👍